Corrfunc/tests/common.py

```diff
@@ -0,0 +1,101 @@
+import os
+from os.path import dirname, abspath, join as pjoin
+
+import numpy as np
+import pytest
+
+from Corrfunc.io import read_fastfood_catalog
+
+@pytest.fixture(scope='module')
+def gals_Mr19():
+    
+    filename = pjoin(dirname(abspath(__file__)),
+                    "../../theory/tests/data", "gals_Mr19.ff")
+    x, y, z, w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return x, y, z, w
+
+
+@pytest.fixture(scope='module')
+def Mr19_mock_northonly():
+    filename = pjoin(dirname(abspath(__file__)),
+                "../../mocks/tests/data", "Mr19_mock_northonly.rdcz.ff")
+    ra,dec,cz,w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return ra, dec, cz, w
+
+
+@pytest.fixture(scope='module')
+def Mr19_randoms_northonly():
+    filename = pjoin(dirname(abspath(__file__)),
+                "../../mocks/tests/data", "Mr19_randoms_northonly.rdcz.ff")
+    ra,dec,cz,w = read_fastfood_catalog(filename, need_weights=True)
+    
+    return ra, dec, cz, w
+
+
+def maxthreads():
+    '''Use as many threads as cores that are available to this process'''
+    import multiprocessing
+    
+    try:
+        maxthreads = len(os.sched_getaffinity(0))
+    except AttributeError:
+        maxthreads = multiprocessing.cpu_count() or 1
+        
+    return maxthreads
+
+
+def generate_isa_and_nthreads_combos(thread_sweep_max=4):
+    '''Test all ISA with maxthreads, and the fastest ISA with threads 1 to `fastest_nthreads`'''
+    mx = maxthreads()
+    
+    # this is the max number of threads we will use when testing multiple values of nthreads
+    # we will cap this at the number of cores
+    thread_sweep_max = min(thread_sweep_max,mx)
+    all_nthreads = list(range(1,thread_sweep_max+1))
+    # ... except for one test, which we'll force to have more threads than cores as a "stress test"
+    all_nthreads += [mx+1]
+    
+    combos = []
+    all_isa = ['fallback','sse42','avx','avx512f']
```

Comment:
`wp` also has an `AVX2` kernel - should we add that in specifically somehow - or just ignore that from python?